src/exportImage.py

```diff
@@ -56,4 +58,14 @@ def create_index(self, filename, fullPath):
         sql = f'CREATE unique INDEX tiles_index on {filename}(zoom_level, tile_column, tile_row)'
         data_source.ExecuteSQL(sql)
 
-
+    def runPreperation(self, taskId, filename):
+        status = self.__helper.get_status(taskId)
+        if status["status"] == Status.COMPLETED.value:
```

Comment:
add todo to change for better check (gdal timeout)

Reply:
should remain regardless 
and the modification for the timeout is quite big and isn't related to this function